src/github/repository.ts

```diff
@@ -1,8 +1,46 @@
 import { emitWebhookProcessedMetrics } from "utils/webhook-utils";
 import { WebhookContext } from "routes/github/webhook/webhook-context";
 import { transformRepositoryId } from "~/src/transforms/transform-repository-id";
+import { Subscription } from "models/subscription";
+import { RepoSyncState } from "models/reposyncstate";
+import { findOrStartSync } from "~/src/sync/sync-utils";
+import { booleanFlag, BooleanFlags } from "config/feature-flags";
 
-export const deleteRepositoryWebhookHandler = async (context: WebhookContext, jiraClient, _util, gitHubInstallationId: number): Promise<void> => {
+
+export const repositoryWebhookHandler = async (context: WebhookContext, jiraClient, _util, gitHubInstallationId: number, subscription: Subscription): Promise<void> => {
+	if (context.action === "deleted") {
+		return deleteRepositoryWebhookHandler(context, jiraClient, gitHubInstallationId);
+	}
+	if (context.action === "created" && await booleanFlag(BooleanFlags.REPO_CREATED_EVENT, jiraClient.baseURL)) {
+		return createRepositoryWebhookHandler(context, gitHubInstallationId, subscription);
+	}
+};
+
+export const createRepositoryWebhookHandler = async (context: WebhookContext, gitHubInstallationId: number, subscription: Subscription): Promise<void> => {
+	const { payload: { repository } } = context;
+	context.log = context.log.child({ gitHubInstallationId });
+
+	try {
+		await RepoSyncState.create({
+			subscriptionId: subscription.id,
+			repoId: repository.id,
+			repoName: repository.name,
+			repoFullName: repository.full_name,
+			repoOwner: repository.owner.login,
+			repoUrl: repository.html_url,
+			repoUpdatedAt: new Date(repository.updated_at)
+		});
+
+		await subscription.update({ totalNumberOfRepos: (subscription.totalNumberOfRepos || 0) + 1 });
+		await findOrStartSync(subscription, context.log, "partial");
+		webhookProcessComplete(context, 200);
+	} catch (err) {
+		context.log.error({ err }, "Error processing create repository webhook");
+		webhookProcessComplete(context, 500);
+	}
+};
+
+export const deleteRepositoryWebhookHandler = async (context: WebhookContext, jiraClient, gitHubInstallationId: number): Promise<void> => {
```

Comment:
Just realise, should we also DELETE the RepoSyncStates as well? And subtract the Subscriptions.totalRepoNumbers?

Reply:
Coz, otherwise, a full restart backfill will trigger the 404 on the deleted Repo.

Reply:
oooooooo Good catch, thats a good point, I just assumed it was whoops.

Sweet I will add that to the story now and do it as a seperate task!